test/parallel/test-dgram-bind-socket-close-before-cluster-reply.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+const common = require('../common');
+const dgram = require('dgram');
+const cluster = require('cluster');
+
+if (cluster.isPrimary) {
+  cluster.fork();
+} else {
+  // When the socket attempts to bind, it requests a handle from the cluster.
+  // Force the cluster to send back an error code.
+  const socket = dgram.createSocket('udp4');
+  cluster._getServer = function(self, options, callback) {
+    socket.close(() => { cluster.worker.disconnect(); });
+    callback(-1);
+  };
```

Comment:
When we create a socket in the cluster worker, it will request a socket from the main process through `cluster._getServer`. If we close the socket before the main process returns, then the `error` event should not be emitted. So i hack the `cluster._getServer` function to test this case.

Reply:
I see the idea behind it. If there is no other way to get this without the monkey patching (like double binding and similar) then I'm fine with it since it's only in the test anyway.